Arrays/script.js

```diff
@@ -312,3 +312,144 @@ const calcAverageHumanAge = ages =>
+2. Find Sarah's dog and log to the console whether it's eating too much or too little. HINT: Some dogs have multiple owners, so you first need to find Sarah in the owners array, and so this one is a bit tricky (on purpose) 🤓
+3. Create an array containing all owners of dogs who eat too much ('ownersEatTooMuch') and an array with all owners of dogs who eat too little ('ownersEatTooLittle').
+4. Log a string to the console for each array created in 3., like this: "Matilda and Alice and Bob's dogs eat too much!" and "Sarah and John and Michael's dogs eat too little!"
+5. Log to the console whether there is any dog eating EXACTLY the amount of food that is recommended (just true or false)
+6. Log to the console whether there is any dog eating an OKAY amount of food (just true or false)
+7. Create an array containing the dogs that are eating an OKAY amount of food (try to reuse the condition used in 6.)
+8. Create a shallow copy of the dogs array and sort it by recommended food portion in an ascending order (keep in mind that the portions are inside the array's objects)
+
+HINT 1: Use many different tools to solve these challenges, you can use the summary lecture to choose between them 😉
+HINT 2: Being within a range 10% above and below the recommended portion means: current > (recommended * 0.90) && current < (recommended * 1.10). Basically, the current portion should be between 90% and 110% of the recommended portion.
+
+TEST DATA:
+
+
+GOOD LUCK 😀
+*/
+
+const dogs = [
+  { weight: 22, curFood: 250, owners: ['Alice', 'Bob'] },
+  { weight: 8, curFood: 200, owners: ['Matilda'] },
+  { weight: 13, curFood: 275, owners: ['Sarah', 'John'] },
+  { weight: 32, curFood: 340, owners: ['Michael'] },
+];
+//1.
+dogs.forEach(function (movs, i) {
+  dogs[i].recommendedFood = Math.trunc(dogs[i].weight ** 0.75 * 28);
+});
+console.log(dogs);
+//2.
+dogs.forEach(function (movs, i) {
+  if (dogs.at(i).owners.includes('Sarah')) {
+    if (dogs.at(i).curFood > dogs.at(i).recommendedFood) {
+      console.log("Sarah's dog eats too much");
+    } else {
+      console.log("Sarah's dog eats too little");
+    }
+  }
+});
+//3.
+const arr1 = [];
+const arr2 = [];
+dogs.map(function (mov, i) {
+  if (dogs.at(i).curFood > dogs.at(i).recommendedFood) {
+    arr1.push(dogs.at(i).owners);
+  } else {
+    arr2.push(dogs.at(i).owners);
+  }
+});
+const ownersEatTooMuch = arr1.flat();
+const ownersEatTooLittle = arr2.flat();
+console.log(ownersEatTooMuch, ownersEatTooLittle);
+//4.
+console.log(`${ownersEatTooMuch.join(' & ')}'s dogs eat too much! `);
+console.log(`${ownersEatTooLittle.join(' and ')}'s dogs eat too little! `);
+//5.
+console.log(
+  dogs.some((el, i) => dogs.at(i).curFood === dogs.at(i).recommendedFood)
```

Comment:
I think `.at` is rather used for rare cases when some optimization is needed, in this case, Jonas' solution is simpler and easier to understand, and same has speed.